src/plugin/customParseFormat/index.js

```diff
@@ -80,7 +43,55 @@ function correctHours(time) {
   }
 }
 
-function makeParser(format) {
+function makeParser(format, instance) {
+  const expressions = {
```

Comment:
@Kreozot 

Everything else looks fine in this PR, and will be merged into next release.

Please move this object to outer scope for a better performance.

ref #441

Reply:
This object contains two properties that depends of current instance locale.
But I can move everything but this two properties and then merge it together. Is it ok?

Reply:
Something like this:
```
  const expressions = {
    ...baseExpressions,
    MMM: [matchWord, function (input) {
      const locale = instance.$locale()
      const { months } = locale
      this.month = months.findIndex(month => month.substr(0, 3) === input) + 1
    }],
    MMMM: [matchWord, function (input) {
      const locale = instance.$locale()
      const { months } = locale
      this.month = months.indexOf(input) + 1
    }]
  }
```

Reply:
It's better to get `this.locale` before `makeParser ` called, and set it to a variable in the outer scope.

Reply:
Don't you think, it is a little bit dirty? For now the only way to change locale is to set it globally. But if it will be changed in future then global variable may cause failure.